frontend/src/components/ShootWorkers/GWorkerInputGeneric.vue

```diff
@@ -512,22 +476,13 @@ export default {
       }
       this.v$.volumeSize.$touch()
     },
-    onInputMinimum () {
-      this.v$.worker.minimum.$touch()
-    },
-    onInputMaximum () {
-      this.v$.worker.maximum.$touch()
-    },
-    onInputMaxSurge () {
-      this.v$.worker.maxSurge.$touch()
-      this.$emit('updateMaxSurge', { maxSurge: this.worker.maxSurge, id: this.worker.id })
-    },
     onInputZones () {
       this.v$.selectedZones.$touch()
       this.v$.worker.maximum.$touch()
     },
     resetWorkerMachine () {
-      this.worker.machine.type = get(head(this.machineTypes), 'name')
+      const machineType = head(this.machineTypes)
+      this.worker.machine.type = get(machineType, 'name')
       const machineImage = head(this.machineImages)
       this.worker.machine.image = pick(machineImage, ['name', 'version'])
```

Comment:
```suggestion
      this.worker.machine.type = machineType?.name
      const machineImage = this.defaultMachineImageForCloudProfileNameAndMachineType(
        this.cloudProfileName,
        machineType,
      )
```
This is not a new bug, also broken in ucrrent master. We need to use the store function to get the correct machine image. Simply using the first one may select a preview / deprecated version